### PR TITLE
Added a PushCommandResource and a v2 and v3 resource providers for them.

### PR DESCRIPTION
**Pull request by deepakaravindr**
Related to https://github.com/NuGet/Home/issues/1202

@yishaigalatzer @emgarten @pranavkm @feiling @johnataylor

### CODE REVIEW

File: src/NuGet.Protocol.Core.Types/Resources/PushCommandResource.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        {
+            if (pushEndpoint == null)
+            {
+                throw new ArgumentNullException(nameof(pushEndpoint));
+            }
+
+            PushEndpoint = pushEndpoint;
+        }
+
+        public Task<string> GetPushEndpointAsync(CancellationToken token)
```
Comment (MeniZalzman) [inline src/NuGet.Protocol.Core.Types/Resources/PushCommandResource.cs:20]: can we change this method to be not async

File: src/NuGet.Protocol.Core.v2/PushCommandResourceV2Provider.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        {
+            PushCommandResource pushCommandResource = null;
+            var v2Repo = await GetRepository(source, token);
+
+            if (v2Repo != null && v2Repo.V2Client != null && !string.IsNullOrEmpty(v2Repo.V2Client.Source))
+            {
+                pushCommandResource = new PushCommandResource(v2Repo.V2Client.Source);
+            }
+
+            return new Tuple<bool, INuGetResource>(pushCommandResource != null, pushCommandResource);
```
Comment (MeniZalzman) [inline src/NuGet.Protocol.Core.v2/PushCommandResourceV2Provider.cs:29]: split the return

File: src/NuGet.Protocol.Core.v3/PushCommandResourceV3Provider.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            if (serviceIndex != null)
+            {
+                var baseUrl = serviceIndex[ServiceTypes.PackagePublish].FirstOrDefault();
+                if (baseUrl != null)
+                {
+                    pushCommandResource = new PushCommandResource(baseUrl.AbsoluteUri);
+                }
+            }
+
+            return new Tuple<bool, INuGetResource>(pushCommandResource != null, pushCommandResource);
```
Comment (MeniZalzman) [inline src/NuGet.Protocol.Core.v3/PushCommandResourceV3Provider.cs:35]: split return

**Comment by MeniZalzman** [comment]
:shipit:

File: src/NuGet.Protocol.Core.Types/Resources/PushCommandResource.cs (hunk 1 above)
Comment (deepakaravindr) [inline src/NuGet.Protocol.Core.Types/Resources/PushCommandResource.cs:20]: Fixed

File: src/NuGet.Protocol.Core.v2/PushCommandResourceV2Provider.cs (hunk 2 above)
Comment (deepakaravindr) [inline src/NuGet.Protocol.Core.v2/PushCommandResourceV2Provider.cs:29]: Fixed

File: src/NuGet.Protocol.Core.v3/PushCommandResourceV3Provider.cs (hunk 3 above)
Comment (deepakaravindr) [inline src/NuGet.Protocol.Core.v3/PushCommandResourceV3Provider.cs:35]: Fixed

**Comment by deepakaravindr** [comment]
Merged with https://github.com/NuGet/NuGet3/commit/e890fc47d4a5cd6f28f964a1319aa09af74a51b3

**Pull request closed without merging**